### PR TITLE
workflows: Include only bazel invocations in trends view

### PR DESCRIPTION
**Pull request by bduffany**
Exclude `ROLE=CI_RUNNER` from trends queries so that we don't double-count invocations (the CI runner invocation should not be counted because it just spawns Bazel invocations).

---

**Version bump**: Patch

### CODE REVIEW

**Comment by siggisim** [comment]
This is cool with me as long as it doesn't mess with the table indexes / performance improvements that @tylerwilliams made. @tylerwilliams  any idea on how we can figure out whether or not it does?

File: enterprise/server/invocation_stat_service/invocation_stat_service.go [hunk 1]
```diff
@@ -104,6 +104,10 @@ func (i *InvocationStatService) GetTrend(ctx context.Context, req *inpb.GetTrend
 
 	if role := req.GetQuery().GetRole(); role != "" {
 		q.AddWhereClause("role = ?", role)
+	} else {
+		// Don't include workflow invocations in trends since workflows just "wrap"
+		// other invocations, and we want to avoid double-counting.
+		q.AddWhereClause(`role != "CI_RUNNER"`)
```
Comment (tylerwilliams) [inline enterprise/server/invocation_stat_service/invocation_stat_service.go:110]: Can the client whitelist what it wants from the request instead? It's not nice to have the server silently messing with this query on an API that makes no mention of it / really shouldn't.

(Also, what is the set of roles we currently know about?)

Separately, we should profile this query to ensure it works for our customers with millions of invocations.

Comment (bduffany) [reply enterprise/server/invocation_stat_service/invocation_stat_service.go]: Updated for the client to send a whitelist of roles.

Did some profiling, for our largest customer it takes 5.8s for this query _without_ any `role = ?` clauses, and around 7.3s for this query with `role = ?` clauses (including our existing `role = "CI"` clause).

Worked with Tyler offline, added a new index which includes the role, and now both types of queries (with or without "role" filters) are just as fast.

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**